pkg/controller/installation/windows/calico_windows_upgrader.go

```diff
@@ -267,6 +286,10 @@ func (w *calicoWindowsUpgrader) Start(ctx context.Context) {
 			case install := <-w.installChan:
 				w.install = install
 			case <-ticker.C:
+				if w.install.KubernetesProvider != operatorv1.ProviderAKS {
```

Comment:
Rather than exit, I think we should do something like this:

```go
if w.install.KubernetesProvider == operatorv1.ProviderAKS {
  w.updateWindowsNodes()
}
```

this has a couple benefits:
- We don't exit the loop - thus, if the provider is manually set to something else and then back to AKS, we're recover (since Start() is only called once at controller startup). 
- We don't need the `w.isRunning` bool - we'll always handle config updates, we'll just not take any action unless we're on AKS.

Reply:
@caseydavenport yes I agree. As it is now, if the user happens to change the installation's `kubernetesProvider` to anything other than AKS, that will effectively turn off the calicoWindowsUpgrader and the only way to get it running again is to restart the operator.

Reply:
5cd4def